psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1931,33 +2221,44 @@ void OEProp::compute_wiberg_lowdin_indices()
 
 //    Print out the bond index matrix
 //    A nicer output is needed ...
-
-    if (same_dens_) {
-        WBI_total->print();
-        outfile->Printf( "  Atomic Valences: \n");
-        WBI_valence->print();
-    }
-    else {
-        outfile->Printf( "  Total Bond Index: \n");
-        WBI_total->print();
-        outfile->Printf( "  Alpha Contribution: \n");
-        WBI_alpha->print();
-        outfile->Printf( "  Beta Contribution: \n");
-        WBI_beta->print();
-        outfile->Printf( "  Atomic Valences: \n");
-        WBI_valence->print();
+    if (print_output)
+    {
+        if (same_dens_) {
+            WBI_total->print();
+            outfile->Printf( "  Atomic Valences: \n");
+            WBI_valence->print();
+        }
+        else {
+            outfile->Printf( "  Total Bond Index: \n");
+            WBI_total->print();
+            outfile->Printf( "  Alpha Contribution: \n");
+            WBI_alpha->print();
+            outfile->Printf( "  Beta Contribution: \n");
+            WBI_beta->print();
+            outfile->Printf( "  Atomic Valences: \n");
+            WBI_valence->print();
+        }
     }
-    wfn_->set_array("WIBERG_LOWDIN_INDICES", WBI_total);
+    return std::make_tuple(WBI_total,WBI_alpha,WBI_beta,WBI_valence);
 }
 
 void OEProp::compute_no_occupations()
+{
+    std::vector<std::vector<std::tuple<double, int, int> >> metrics;
+    pac.compute_no_occupations(metrics, max_noon_,true);
+    wfn_->set_no_occupations(metrics);
+}
+void PopulationAnalysisCalc::compute_no_occupations(std::vector<std::vector<std::tuple<double, int, int> > > & output_metrics, int max_noon, bool print_output)
```

Comment:
Looking at other parts of the code, would it make more sense to return the output_metrics rather than have to build them and pass them in?

Reply:
It would be very nice to have this as an actual return value.
The reason i did not do it yet is simply:
The return type would be 
std::shared_ptr<std::vector<std::vector<std::tuple<double, int, int> > > >
which I found quite excessive. Everything else would lead to extra copies.

Reply:
Hmm, this way a user can write `auto my_pac-> compute_no_occupations(5, true);` which would be more natural, I think. Either way.

Reply:
I did not consider the auto keyword. You are absolutely right and I will change this immediately, because it is so much cleaner.